default.nix

```diff
@@ -50,7 +51,34 @@ let
 
       # Filter packages by their PEP508 markers
       partitions = let
-        supportsPythonVersion = pkgMeta: if pkgMeta ? marker then (evalPep508 pkgMeta.marker) else true;
+
+        supportsPythonVersion = pkgMeta: let
```

Comment:
I'm fine with the change per se but I'd very much like for the code below to be split up and moved to `lib.nix`. I find that this stack of let bindings kind of hides the semantics.

```nix
# Returns true if ...
supportsPythonVersion = pkgMeta: pyProject:
```

Looking at `lib.nix` I also see that the identical `combine` binding is over there as well so this could also be extracted. So :+1: on all of this but would be great if it could be refactored a bit such that machinery like `combine` and the logic of `the whole check yielding true weren't all happening on the same level which makes this harder to parse and follow - for my silly brain anyhow :sweat_smile:

Reply:
Yeah they were pretty much identical, I've unified the logic.